src/core/client.ts

```diff
@@ -9,7 +11,15 @@ import { LinkedInRequest } from './linkedin-request';
 import { Login } from './login';
 
 export class Client {
-  request = new LinkedInRequest();
+  request: LinkedInRequest;
+
+  constructor();
+  constructor(proxy: AxiosProxyConfig);
```

Comment:
What's the reason for method overloading here? Why not just `constructor(proxy?: AxiosProxyConfig)` ?